scio-core/src/main/java/com/spotify/scio/transforms/AsyncLookupDoFn.java

```diff
@@ -30,29 +46,13 @@
  * @param <B> client lookup value type.
  * @param <C> client type.
  */
-public abstract class AsyncLookupDoFn<A, B, C> extends DoFn<A, KV<A, AsyncLookupDoFn.Try<B>>> {
+public abstract class AsyncLookupDoFn<A, B, C> extends AsyncDoFn<A, B, C,
+    AsyncLookupDoFn.Try<B>, ListenableFuture<B>> {
   private static final Logger LOG = LoggerFactory.getLogger(AsyncLookupDoFn.class);
 
-  // DoFn is deserialized once per CPU core. We assign a unique UUID to each DoFn instance upon
-  // creation, so that all cloned instances share the same ID. This ensures all cores share the
-  // same Client and Cache.
-  private static final ConcurrentMap<UUID, Object> client = Maps.newConcurrentMap();
-  private static final ConcurrentMap<UUID, Cache> cache = Maps.newConcurrentMap();
-  private final UUID instanceId;
-
-  private final CacheSupplier<A, B, ?> cacheSupplier;
-
   // Data structures for handling async requests
-  private final Semaphore semaphore;
   private final ConcurrentMap<UUID, ListenableFuture<B>> futures = Maps.newConcurrentMap();
   private final ConcurrentLinkedQueue<Result> results = Queues.newConcurrentLinkedQueue();
```

Comment:
Is `Result` the `case class Result` in Scala? In that case this Java class is hard to use in pure Java Beam. Also it could be abstracted away as a type param in parent, or just a pure Java class.

Reply:
Result is Java, private Result class in AsyncLookupDoFn